### PR TITLE
Hotfix | Prerendering

### PR DESCRIPTION
**Pull request by ItsMeBrianD**
### Checklist

- [ ] ~~For UI or styling changes, I have added a screenshot or gif showing before & after~~
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [ ] ~~I have added to the docs where applicable~~
- [ ] ~~I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable~~

### CODE REVIEW

File: packages/lib/preprocess/src/process-queries.cjs [hunk 1]
```diff
@@ -42,79 +42,81 @@ const createDefaultProps = function (filename, componentDevelopmentMode, duckdbQ
...
+                let ${id};
+
+                $: __${id}HasUnresolved = hasUnsetValues\`${duckdbQueries[id].compiledQueryString.replaceAll('`', '\\`')}\`;
+                $: __${id}Text = \`${duckdbQueries[id].compiledQueryString.replaceAll('`', '\\`')}\`
+
+                // keep initial state around until after the query has resolved once
+                let __${id}InitialFactory = false;
+                $: if (__${id}HasUnresolved || !__${id}InitialFactory) {    
+                    if (!__${id}HasUnresolved) {
+                        __${id}Factory(__${id}Text, { noResolve: __${id}HasUnresolved, ...${id}InitialStates });
```
Comment (csjh) [inline packages/lib/preprocess/src/process-queries.cjs:97]: not 100% sure this part was required for the fix, need to check

File: packages/lib/sdk/src/usql/query/Query.js [hunk 2]
```diff
@@ -367,8 +367,12 @@ ${this.text.trim()}
...
 		return resolved;
 	};
-	fetch = async () => {
-		return Promise.allSettled([this.#fetchColumns(), this.#fetchData()]).then(() => this.value);
+	fetch = () => {
+		const cols = this.#fetchColumns();
+		if (cols instanceof Promise) {
+			return Promise.allSettled([this.#fetchColumns(), this.#fetchData()]).then(() => this.value);
+		}
+		this.#fetchData();
```
Comment (csjh) [inline packages/lib/sdk/src/usql/query/Query.js:375]:
```suggestion
		this.#fetchData();
		return Promise.resolve();
```
I think we should return `Promise.resolve()` so people can assume the return value is thenable

File: packages/lib/sdk/src/usql/query/Query.js [hunk 3]
```diff
@@ -367,8 +367,12 @@ ${this.text.trim()}
 		);
 		return resolved;
 	};
-	fetch = async () => {
-		return Promise.allSettled([this.#fetchColumns(), this.#fetchData()]).then(() => this.value);
+	fetch = () => {
+		const cols = this.#fetchColumns();
+		if (cols instanceof Promise) {
+			return Promise.allSettled([this.#fetchColumns(), this.#fetchData()]).then(() => this.value);
```
Comment (csjh) [inline packages/lib/sdk/src/usql/query/Query.js:373]:
```suggestion
			return Promise.allSettled([this.#fetchColumns(), this.#fetchData()]);
```
either return `this.value` in the sync version or stop returning `this.value` in either path

File: packages/lib/sdk/src/usql/query/Query.js (hunk 2 above)
Comment (ItsMeBrianD) [reply packages/lib/sdk/src/usql/query/Query.js:375]: This means that `instanceof Promise` will always be true though, so the current way we check for async code could no longer wrok

**Review by csjh — APPROVED** [review]

**Pull request merged**